src/stores/auth.js

```diff
@@ -117,9 +117,14 @@ export const useAuthStore = defineStore('auth', {
           const { gameId } = route.params;
           gameStore.requestSpectate(Number(gameId));
         }
-
+        
         if (gameId && (isGame || isLobby)) {
           await this.requestReauthenticate({ username }).then(({ game }) => {
+            if (game.status === GameStatus.FINISHED) {
+              gameStore.updateGame(game.lastEvent.game);
+              gameStore.setGameOver(game.lastEvent.victory);
+              return;
+            }
```

Comment:
Okay I see now that we're not emitting the event for the gameover case, we're just using the returned game from the response to the reLogin request. That has the nice property of not bothering the other user who is likely still connected and doesn't need the socket event.

The only concern I have is whether there is cleanup logic in the socket event handlers that's missed by calling updateGame and setGameOver w/o firing the actual socket event handlers themself e.g. other variables that need resetting.